src/main/java/com/flowingcode/vaadin/addons/twincolgrid/TwinColGrid.java

```diff
@@ -84,9 +84,9 @@
 
 	private final TwinColModel<T> right = new TwinColModel<>();
 
-	protected final Grid<T> leftGrid = left.grid;
+	protected final Grid<T> leftGrid;
 
-	protected final Grid<T> rightGrid = right.grid;
+	protected final Grid<T> rightGrid;

```

Comment:
Make these final fields static too
[https://sonarcloud.io/organizations/flowingcode/rules?open=java%3AS1170&rule_key=java%3AS1170](url)

Reply:
These fields must NOT be static (since both of them depends on `left`/`right`, which aren't static). For some reason Sonar doesn't realizes it.

Reply:
Moved initialization of `left`/`right` into the constructor too.